app/controllers/bills_controller.rb

```diff
@@ -0,0 +1,43 @@
+class BillsController < ApplicationController
+
+  def new
+    set_book
+    set_product 
+    @session = Stripe::Checkout::Session.create({
+      payment_method_types: ['card'],
+      line_items: [
+        price_data: {
+          product: @product.id,
+          unit_amount: (@book.price * 100).to_i,
+          currency: 'usd',
+        },
+        quantity: 1,
+      ],
+      mode: 'payment',
+      success_url: books_url,
+      cancel_url: book_url(@book.id),
+    })
+    respond_to do |format|
+      format.js
+    end
+  end
+
+  private
+    def set_book
+      @book = Book.find(params[:book])
+    end
+
+    def set_product
+      if @book.product_id.nil?
+        @product = Stripe::Product.create({
+                    name: @book.title,
+                    images: [@book.image_link],
+                    description: "Description: " + @book.description
+                  })
```

Comment:
Is there any kind of exceptions thrown from the create method that we should handle?